TST use global_random_seed in sklearn/linear_model/tests/test_linear_loss.py

sortofamudkip opened the pull request:
<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/main/CONTRIBUTING.md
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->
Towards #22827. All tests passed. @adrinjalali 

#### What does this implement/fix? Explain your changes.

Added `global_random_seed` parameter to: 
* `test_init_zero_coef`
* `test_loss_grad_hess_are_the_same` (passed to `random_X_y_coef`)
* `test_loss_gradients_hessp_intercept` (passed to `random_X_y_coef`)
* `test_gradients_hessians_numerically` (passed to `random_X_y_coef`)
* `test_multinomial_coef_shape` (passed to `random_X_y_coef`)
* `test_multinomial_hessian_3_classes` (passed to `random_X_y_coef`)

<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
https://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

sortofamudkip committed:
add global_random_seed in tests

StefanieSenger commented:
Hi @sortofamudkip,

thanks for your contribution. In order to properly run all tests, can you please push an empty commit inlcuding all the names of the tests that you have changed, like this:

git commit --allow-empty -m "empty commit to trigger all CI jobs [all random seeds]"
test_1
test_2
test_3
name_of_more_tests
"

This is mentioned in the issue an d needed to run the tests with all the available seeds.

sortofamudkip committed:
replace 42 with global_random_seed in two places [all random seeds]
test_init_zero_coef
test_loss_grad_hess_are_the_same
test_loss_gradients_hessp_intercept
test_gradients_hessians_numerically
test_multinomial_coef_shape
test_multinomial_hessian_3_classes

sortofamudkip committed:
remove modified tests from test_regression.py [all random seeds]
test_init_zero_coef
test_loss_grad_hess_are_the_same
test_loss_gradients_hessp_intercept
test_gradients_hessians_numerically
test_multinomial_coef_shape
test_multinomial_hessian_3_classes

StefanieSenger reviewed (approved):
Thank you @sortofamudkip, it looks all good to me.

I think we can merge this. Would you mind having a look @ogrisel or @betatim? This is from the sprint we had in Berlin yesterday.

OmarManzoor reviewed (approved):
LGTM. Thanks @sortofamudkip
